### ISSUE TITLE
Santander Customer Transaction Predict (Kaggle 과제) 분석

### ISSUE THREAD

**Issue by memeoo** [gema0000/kerasStudy2#2]
Santander Customer Transaction Predict
-- 문제의 분석 
* Training 
X : var0 ~ var199의 200개 DataSet(Column)의 20만개의 데이터(Row) 
Y : 0 혹은 1.

 * Test
X: var0 ~ var199의 200개 DataSet(Column)의 20만개의 데이터(Row) 
Y: 0 혹은 1 을 맞추기

Santander는 아마 은행인 듯 싶습니다.
var0 부터 var 199까지의 x 변수는 고객에 관련된 변수(나이, 예금액, 성별, 신용카드 여부, 연체율 등등...)로 추정되고, 
Y 변수는 고객의 이탈율과 같은 변수로 추정됩니다. (무엇인지는 나와 있지는 않고- 보안이 필요하겠죠, 중요하지도 않습니다.)
X 변수들은 특정한 기준으로 scaled 된것 같습니다.(+두자리.소수점 ~ - 두자리.소수점)
아무튼 var0부터 var199까지의 200개의 고객 데이터를  가지고 0,1 두 값으로 매핑 하는 분류 문제로 보면 될것 같습니다.
 
- 문제의 접근
* X data가 scaled 되었기 때문에, 추가적인 data 가공은 필요치 않아 보입니다.
* 따라서 Pandas를 이용해 traninX, trainY, testX, testY 데이터셋를 만들어 케라스로 모델링 하면 될것 같습니다.

**Comment by memeoo** [comment]
지금 다시 확인해 보니 test.csv 파일은 target(결과값)을 주지 않았네요.
즉, train.csv 의 데이터를 70%, 30% 정도로 쪼개서 70% 정도를 학습 데이터로, 나머지 30% 데이터를 Test 및 Validataion 용도로 써야 할것 같습니다. 
그러면 test.csv 파일의 용도가 좀 궁금하네요. 예측값(predic result)을 test.csv 에다 채워 넣으라는 의도인지...;;
캐글을 해본적이 없어서 어떤 방식으로 돌아가는지 확신이 안 서내요.

**Comment by memeoo** [comment]
import numpy as np
import pandas as pd 
from keras.layers import Dense, Activation, Flatten 
from keras.layers.core import Flatten
from keras.models import Sequential, Model

trainCsv = pd.read_csv("./kaggle/train.csv")

xTrain = trainCsv.iloc[:140000, 2:]
yTrain = trainCsv.target[:140000]

xTest = trainCsv.iloc[140000:, 2:]
yTest = trainCsv.target[140000:]

model = Sequential()
model.add(Dense(100, input_shape=(200,), activation='sigmoid'))
model.add(Dense(32, activation='sigmoid'))
model.add(Dense(120))
model.add(Dense(64, activation='sigmoid'))
model.add(Dense(1, activation='sigmoid'))
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['acc'])

model.summary()
model.fit(xTrain, yTrain, epochs=10, batch_size=64, validation_data=(xTest, yTest))

a, b = model.evaluate(xTest, yTest, batch_size=64)
print(a, b)

y_predict = model.predict(xTest)
print(y_predict)

이렇게 해봤더니, acc=0.90748 정도 나오네요..ㅎㅎ;
